dart-templates/serialization/native/class.mustache

```diff
@@ -251,7 +251,14 @@ class {{{classname}}} {{#interfaces}}{{#-first}}implements {{/-first}}{{.}}{{^-l
     return <String, dynamic>{
   {{#vars}}
     {{^required}}
-    if ((keys == null && {{{name}}} != null) || (keys?.contains(r'{{{name}}}') ?? false))
+    if (keys == null || keys.
+    {{#isModel}}
+    any((key) => RegExp(r'^{{{name}}}.').hasMatch(key))
```

Comment:
```suggestion
    any((key) => key == r'{{{name}}}' || key.sartswith(r'{{{name}}}.')
```
because the dot in regexp coul mean any character and 'adress2' would be valid for 'adress' key

Reply:
should I just escape it ? `any((key) => RegExp(r'^{{{name}}}\.').hasMatch(key))`
or do we want to accept `adress` as a valid key ? If so, does that mean all of its properties should be serialized ?

Reply:
Hmm you might be right, I just wonder if there are some cases where accepting all properties could be useful
We can go ahead with the escaped dot